mesonbuild/interpreter.py

```diff
@@ -650,6 +651,12 @@ def run_method(self, args, kwargs):
     def get_id_method(self, args, kwargs):
         return self.compiler.get_id()
 
+    def get_unittest_flag_method(self, args, kwargs):
+        # At time, only D compilers have this feature.
+        if not getattr(self.compiler, "get_unittest_flag", None):
+            return ''
```

Comment:
Use `hasattr` instea of `getattr`, it is more readable. And throw an exception instead of returning an empty string.